### ISSUE TITLE
Unable to listen with error in rdma_bind_addr for test/examples/ucp_client_server

### ISSUE THREAD

**Issue by Akshay-Venkatesh** [openucx/ucx#3139]
Hi,

I'm running on master branch:
```shell
$ git log | head
commit 7568aec3503e00989b4a4074640c3f19194ab976
Merge: f4cd8ee 80d9ef4
Author: Yossi Itigin <yosefe@mellanox.com>
Date:   Wed Jan 16 01:05:55 2019 +0800

    Merge pull request #3134 from hoopoepg/topic/typo-in-ud-inl-comment
    
    UD: fixed typo in comment

commit 80d9ef45cc80e7d323cfb3f2a566fb00fcd205ba
```
There are two issues I see:
1. On a machine where this test works correctly, I see this in 
```
$ ucx_info -d | less
...
# Memory domain: rdmacm
#            component: rdmacm
#           supports client-server connection establishment via sockaddr
#   < no supported devices found >
#
```
This is not expected output, right?

2. On another machine where this example fails, I see the same output from ucx_info as above but I also see this when I try to start the server:

```shell
[1547681254.659331] [dgx05:81859:0]      rdmacm_md.c:158  UCX  DEBUG rdma_bind_addr(addr = 0.0.0.0:6452) failed: Cannot assign requested address
[1547681254.659353] [dgx05:81859:0]   ucp_listener.c:263  UCX  ERROR none of the available transports can listen for connections on 0.0.0.0:6452
failed to listen (Destination is unreachable)
failed to start server
```

Any thoughts on why this error occurs? All of udaddy, rdma_server/rdma_client, ucmatose, and rping work.

```shell
$ udaddy -s $dgx05ip
udaddy: starting client
udaddy: connecting
initiating data transfers
receiving data transfers
data transfers complete
test complete
return status 0
$ rdma_client -s $dgx05ip
rdma_client: start
rdma_client: end 0
```

It also seems like 0.0.0.0:6452 is valid because for the first case which passes I see the following:

```shell
[1547682603.185289] [hsw210:11491:0]      rdmacm_md.c:181  UCX  DEBUG address 0.0.0.0:6452 (port 6452) is accessible from rdmacm_md 0x23e0380 with mode: 0
[1547682603.185392] [hsw210:11491:0]   rdmacm_iface.c:534  UCX  DEBUG rdma_cm id 0x2485640 listening on 0.0.0.0:6452:6452
```

**Comment by evgeny-leksikov** [comment]
@Akshay-Venkatesh < no supported devices found > is OK for rdmacm memory domain because it's not bound to any device. I agree that message looks confusing and should be fixed to smth else.
Could you check if 6452 port is not used on your second machine? for example, by other instance of test.

**Comment by Akshay-Venkatesh** [comment]
You were right @evgeny-leksikov 
Thanks for the help!

**Comment by Akshay-Venkatesh** [comment]
@evgeny-leksikov One suggestion is to throw a better error for this case. Would it be possible to detect that port is in use?

**Comment by Akshay-Venkatesh** [comment]
@evgeny-leksikov Another thing I wanted to check is why ucx doesn't retry another port number to bind to.  Is it the case that `6452` is reserved? I checked `ucp_context.c` to see if there is an environment variable to set this but it doesn't seem to be the case. Lmk if you know more.

**Comment by Akshay-Venkatesh** [comment]
Hi, Yossi.

When I wasn't able to run, I wasn't able to run the server at all. I simply compiled `test/examples/ucp_client_server.c` using `gcc ucp_client_server.c -o ucp_client_server -L $PWD/lib -lucm -lucs -lucp -luct ` and ran `./ucp_client_server`. This lead to the error. So by default rdma_bind_addr is attempting to bind at `0.0.0.0:6452`. This seems external to the 13337 port number chosen by the test as chosen by `-p` argument. What are your thoughts on this?

**Comment by yosefe** [comment]
default port selection in client/server example fixed in #3533 
error handling of busy port fixed in #3477
